src/python/genny/genny_auto_tasks.py

```diff
@@ -45,17 +80,83 @@ def modified_workload_files():
-    except subprocess.CalledProcessError as e:
-        print(e.output, file=sys.stderr)
-        raise e
 
-    return out.decode().strip()
+    # First check that the workload has a proper AutoRun section.
+    if 'AutoRun' not in workload_dict or not isinstance(workload_dict['AutoRun'], dict):
+        return False
+    if 'Requires' not in workload_dict['AutoRun'] or not isinstance(workload_dict['AutoRun']['Requires'], dict):
+        return False
+
+    for module, required_config in workload_dict['AutoRun']['Requires'].items():
+        if module not in env_dict:
+            return False
+        if not isinstance(required_config, dict):
+            return False
+
+        # True if set of config key-vaue pairs is subset of env_dict key-value pairs
+        # This will be false if the AutoRun yaml uses a logical or (i.e. branch_name: master | prod), but it is efficient so we use it for a first-pass.
+        if not required_config.items() <= env_dict[module].items():
+            # Now have to check all k, v pairs individually
+            for k, v in required_config.items():
+                if k not in env_dict[module]:
+                    return False
+                if v != env_dict[module][k]:
+                    if '|' not in v:
+                        return False
+                    
+                    found_match = False
+                    for expr in v.split('|'):
+                        if expr.strip() == env_dict[module][k]:
+                            found_match = True
+                            break
+
+                    if not found_match:
+                        return False
+
+    return True
+
+def autorun_workload_files(env_dict):
+    """
+    :param dict env_dict: a dict representing the values from bootstrap.yml and runtime.yml -- the output of make_env_dict().
+    :return: a list of workload files whose AutoRun critera are met by the env_dict.
+    """
+    workload_dir = '{}/src/workloads'.format(get_project_root())
+    candidates = glob.glob('{}/**/*.yml'.format(workload_dir), recursive=True)
+
+    matching_files = []
+    for fname in candidates:
+        with open(fname, 'r') as handle:
+            try:
+                workload_dict = yaml.safe_load(handle)
+            except Exception as e:
+                continue
+            if workload_should_autorun(workload_dict, env_dict):
+                matching_files.append(fname)
+
+    return matching_files
+
+
+def make_env_dict():
+    """
+    :return: a dict representation of bootstrap.yml and runtime.yml in the cwd, with top level keys 'bootstrap' and 'runtime'
+    """
+    env_files = ['bootstrap.yml', 'runtime.yml']
+    env_dict = {}
+    for fname in env_files:
+        if not os.path.isfile(fname):
+            return None
+        with open(fname, 'r') as handle:
+            config = yaml.safe_load(handle)
+            module = os.path.basename(fname).split('.yml')[0]
+            env_dict[module] = config
```

Comment:
It's better to use an explicitly typed class (or at least a namedtuple) over a dictionary whenever possible. E.g. for `env_dict`